modules/login/src/main/frontend/src/login/loginForm.js

```diff
@@ -33,55 +35,39 @@ import VisibilityIcon from '@mui/icons-material/Visibility';
 import VisibilityOffIcon from '@mui/icons-material/VisibilityOff';
 import styles from "../styling/styles";
 
-class SignIn extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      passwordIsMasked: false,
-      failedLogin: undefined,
-
-      username: "",
-      password: "",
-
-      phase: "USERNAME_ENTRY",
-
-      singleStepEntry: undefined
-    };
-
+function SignIn(props) {
+  checkPropTypes(SignIn, props);
+  const { classes, handleLogin, redirectOnLogin } = props;
+
+  const [ username, setUsername ] = useState("");
+  const [ password, setPassword ] = useState("");
+  const [ failedLogin, setFailedLogin ] = useState(false);
+  const [ passwordIsMasked, setPasswordIsMasked ] = useState(false);
+  const [ phase, setPhase ] = useState("USERNAME_ENTRY");
+  const [ singleStepEntry, setSingleStepEntry ] = useState(undefined);
+ 
+  useEffect(() => {
     // Check to see if 1 or 2 step login should be used
     fetch(window.location.origin + "/apps/cards/SAMLDomains.json")
-    .then((resp) => {
-      if (resp.ok) {
-        this.setState({singleStepEntry: false});
-      } else {
-        this.setState({singleStepEntry: true});
-      }
-    });
-  }
+      .then((resp) => setSingleStepEntry(!!resp.ok));
```

Comment:
This does the opposite than before.
```suggestion
      .then((resp) => setSingleStepEntry(!resp.ok));
```